src/google/protobuf/compiler/csharp/csharp_helpers.cc

```diff
@@ -378,16 +378,32 @@ std::string GetFieldConstantName(const FieldDescriptor* field) {
   return GetPropertyName(field) + "FieldNumber";
 }
 
+// Names of members declared or overridden in the message.
+// Used in GetPropertyName below.
+static const auto& reserved_member_names = *new std::unordered_set<std::string>({
```

Comment:
unfortunately, you are doing this at the global scope which leads to issues.  You need to do this at function scope

Reply:
Fixed, I think - but please check carefully! I'm hoping that by declaring it with the `static` modifier it's still only going to be initialized on first use rather than on every call, right?

Reply:
correct